engine/vcs/gitlab/client_status.go

```diff
@@ -81,17 +81,25 @@ func (c *gitlabClient) SetStatus(ctx context.Context, event sdk.Event) error {
 		Description: &data.desc,
 	}
 
-	if _, _, err := c.client.Commits.SetCommitStatus(data.repoFullName, data.hash, opt); err != nil {
-		return sdk.WrapError(err, "Cannot process event %v - repo:%s hash:%s", event, data.repoFullName, data.hash)
+	val, _, _ := c.client.Commits.GetCommitStatuses(data.repoFullName, data.hash, nil)
+	found := false
+	for _, s := range val {
+		if (s.TargetURL == *opt.TargetURL && s.Status == string(opt.State) && s.Ref == *opt.Ref && s.SHA == data.hash && s.Name == *opt.Name && s.Description == *opt.Description) {
```

Comment:
a check on TargetURL & SHA should be enough, no?

Reply:
We also have to check status (to avoid duplicated). I removed other fields

Reply:
Actually, we must check all of them. After having removed the others fields, the bug appears again